std/algorithm/sorting.d

```diff
@@ -1698,17 +1698,26 @@ private void shortSort(alias less, Range)(Range r)
                 auto t = r[0]; if (pred(t, r[0])) r[0] = r[0];
             }))) // Can we afford to temporarily invalidate the array?
         {
+            import std.algorithm.mutation : move;
+
             size_t j = i + 1;
             auto temp = r[i];
```

Comment:
Is `move` also needed here?

Reply:
Think so. Fixed

Reply:
It does look like it needs to be a `move`. And, since this yanks `r[i]` and moves it at the back, the following loop + fixup can all be `moveEmplace`s, to avoid unnecessary destructor calls. I'd almost vote for `moveEmplaceAndDontInitSource`, but we don't have that :)